metomi/rose/macros/rule.py

```diff
@@ -388,8 +387,13 @@ def _process_rule(
         sci_num_count = -1
 
         # any/all processing.
+        rule = rule.replace('template variables', 'template_variables')
         for array_func_key, rec_regex in self.REC_ARRAY.items():
             for search_result in rec_regex.findall(rule):
+                search_result = (
+                    i.replace('template_variables', 'template variables')
+                    for i in search_result
+                )
```

Comment:
Maybe add a comment explaining why you are replacing one thing one way and then doing the opposite for the other